CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+# We need 3.12 or later, so that we can set policy CMP0074; see below.
+cmake_minimum_required(VERSION 3.12)
+
+# We want find_path() and find_library() to honor {packagename}_ROOT
+if(POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)
+endif()
+
+# MAIN_PROJECT CHECK
+set(PCAPPP_MAIN_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(PCAPPP_MAIN_PROJECT ON)
+endif()
+
+# Include git module required for version
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")
+include(VersionFromGit)
+
+# Get the git version
+version_from_git(
+  LOG
+  ON
+  TIMESTAMP
+  "%Y%m%d%H%M%S")
+
+project(
+  PcapPlusPlus
+  DESCRIPTION "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets."
+  VERSION ${VERSION})
+set(CMAKE_PROJECT_HOMEPAGE_URL "https://pcapplusplus.github.io/")
+
+# Include the other modules
+include(CMakePackageConfigHelpers)
+include(DetectCompiler)
+include(DetectOS)
+include(GNUInstallDirs)
+set(PCAPP_INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR})
+set(PCAPP_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR}/pcapplusplus)
+set(PCAPP_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/pcapplusplus)
+
+# Build options (Turn on Examples and Tests if it's the main project)
+option(PCAPPP_BUILD_EXAMPLES "Build Examples" ${PCAPPP_MAIN_PROJECT})
+option(PCAPPP_BUILD_TESTS "Build Tests" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+
+option(PCAPPP_USE_DPDK "Setup PcapPlusPlus with DPDK. In this case you must also set DPDK_HOME")
+option(PCAPPP_USE_PF_RING "Setup PcapPlusPlus with PF_RING. In this case you must also set PF_RING_HOME")
```

Comment:
ditto: I think this should be `PF_RING_ROOT` and not `PF_RING_HOME`

Reply:
yes correct